### PR TITLE
ARMED support for TMC UART, probe temp

### PR DESCRIPTION
**Pull request by ktand**
### Description

Add TEMP_PROBE_PIN
Add TMC2208/2209 Software Serial Pins

### Benefits

Allows the use of a PINDA2 probe
Allows the use of TMC2208/2209 steppers

### CODE REVIEW

File: Marlin/src/pins/stm32/pins_ARMED.h [hunk 1]
```diff
@@ -117,6 +117,7 @@
 #define TEMP_0_PIN         PC0   // Analog Input
 #define TEMP_1_PIN         PC1   // Analog Input
 #define TEMP_BED_PIN       PC2   // Analog Input
+#define TEMP_PROBE_PIN     PC1   // Analog Input
```
Comment (thisiskeithb) [inline Marlin/src/pins/stm32/pins_ARMED.h]: Take a look at `pins_BTT_SKR_common.h` for how this is handled so you don't end up defining `PC1` to both `TEMP_1_PIN` and `TEMP_PROBE_PIN`: https://github.com/MarlinFirmware/Marlin/blob/b14c933171b50c87808367bf14872fb7acced92d/Marlin/src/pins/lpc1768/pins_BTT_SKR_common.h#L66-L68

Comment (ktand) [reply Marlin/src/pins/stm32/pins_ARMED.h]: Thanks for the suggestion. I'll update the PR.

**Comment by ghost** [comment]
Shouldn't you use "HAS_TMC_UART" instead ? Because defining RX/TX is not needed with other drivers or standalone tmc's ?

**Comment by ktand** [comment]
> Shouldn't you use "HAS_TMC_UART" instead ? Because defining RX/TX is not needed with other drivers or standalone tmc's ?

Thanks. Fixed now.

**Comment by thinkyhead** [comment]
Thanks! I've now applied `HAS_TMC_UART` to the rest of the pins files too.

**Pull request merged**